File: examples/ruby/fixed-array-index/ArrayIndex.runar.rb
# ArrayIndex -- Ruby port of
# examples/ts/fixed-array-index/ArrayIndex.runar.ts.
#
# Exercises FixedArray[Bigint, 4] together with a RUNTIME index read
# @table[i].

require 'runar'

class ArrayIndex < Runar::SmartContract
  prop :table, FixedArray[Bigint, 4], readonly: true, default: [10, 20, 30, 40]

  def initialize
    super()
  end

  runar_public i: Bigint, expected: Bigint
  def lookup(i, expected)
    assert @table[i] == expected
  end
end

File: examples/python/fixed-array-index/ArrayIndex.runar.py
from runar import (
    SmartContract, Bigint, FixedArray, Readonly,
    public, assert_,
)


class ArrayIndex(SmartContract):
    # Python port of examples/ts/fixed-array-index/ArrayIndex.runar.ts.
    #
    # Exercises FixedArray[Bigint, 4] together with a RUNTIME index read
    # self.table[i].

    table: Readonly[FixedArray[Bigint, 4]] = [10, 20, 30, 40]

    def __init__(self):
        super().__init__()

    @public
    def lookup(self, i: Bigint, expected: Bigint):
        assert_(self.table[i] == expected)
